### PR TITLE
[AF-785] accept object in invoke

### PR DESCRIPTION
**Pull request by ocke**
:v:

/cc @zendesk/vegemite

### Description
Accept object in invoke!

### References
* JIRA: https://zendesk.atlassian.net/browse/AF-785

### Risks

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -514,13 +514,25 @@ Client.prototype = {
...
-    var obj = key;
+    var obj = {};
 
     if (typeof key === 'string') {
-      obj = {};
       obj[key] = Array.prototype.slice.call(arguments, 1);
+
+    } else if (typeof key === 'object') {
+      // Validate object
+      Object.keys(key).forEach(function(s) {
```
Comment (svizzari) [inline lib/client.js]: `s` isn't an informative argument name 😾

File: spec/client_spec.js [hunk 2]
```diff
@@ -794,6 +794,33 @@ describe('Client', function() {
             expect(lastCall.appGuid).to.equal('ABC123');
             expect(lastCall.instanceGuid).to.equal('def-321');
           });
+
+          it('makes a call with an object', function() {
+            promise = childClient.invoke({'popover': ['hide']});
```
Comment (svizzari) [inline spec/client_spec.js]: Do we need a test that has an unquoted key name?

Comment (ocke) [inline spec/client_spec.js]: that's the same thing in js

File: lib/client.js [hunk 3]
```diff
@@ -514,13 +514,25 @@ Client.prototype = {
   },
 
   invoke: function(key) {
```
Comment (danielbreves) [inline lib/client.js]: rename to `keyOrObject`?

File: lib/client.js (hunk 1 above)
Comment (danielbreves) [inline lib/client.js]: What's `s`?

File: lib/client.js [hunk 4]
```diff
@@ -514,13 +514,25 @@ Client.prototype = {
...
+    var obj = {};
 
     if (typeof key === 'string') {
-      obj = {};
       obj[key] = Array.prototype.slice.call(arguments, 1);
+
+    } else if (typeof key === 'object') {
+      // Validate object
+      Object.keys(key).forEach(function(s) {
+        var values = key[s];
```
Comment (danielbreves) [inline lib/client.js]: What's `values`? `methodArgs`?

File: lib/client.js [hunk 5]
```diff
@@ -514,13 +514,25 @@ Client.prototype = {
...
+
+        if (!Array.isArray(values) || values.some(function(s) {return typeof s !== 'string'; })) {
```
Comment (danielbreves) [inline lib/client.js]: Can we extract the second condition to a separate function?

File: lib/client.js [hunk 6]
```diff
@@ -514,13 +514,25 @@ Client.prototype = {
...
+          throw new Error('Invoke supports string arguments or an object with array of strings.');
+        }
+      });
+
+      obj = key;
+
     } else {
-      throw new Error('Invoke only supports string arguments.');
+      throw new Error('Invoke supports string arguments or an object with array of strings.');
```
Comment (danielbreves) [inline lib/client.js]: Can we extract this error message to a `var`?

File: lib/client.js (hunk 4 above)
Comment (ocke) [inline lib/client.js]: yeah i guess so

File: spec/client_spec.js (hunk 2 above)
Comment (svizzari) [inline spec/client_spec.js]: > that's the same thing in js

yea, using quoted keys seems unconventional, why not just use unquoted

**Comment by danielbreves** [comment]
@svizzari 💇‍♂️

**Pull request merged**